test/unit/vl-wizard-pane.test.html

```diff
@@ -0,0 +1,184 @@
+        assert.equal(pane.title, pane.querySelector('[slot="title"]').innerText);
+      });
+
+      test('bevat een grid met een content en acties element', () => {
+        const pane = fixture('vl-wizard-pane-fixture');
+        const grid = pane.shadowRoot.querySelector('[is="vl-grid"]');
+        should.exist(grid);
+        assert(grid.hasAttribute('is-stacked'));
+        const columns = grid.querySelectorAll('[is="vl-column"]');
+        assert.lengthOf(columns, 2);
+        columns.forEach(column => assert.equal(column.getAttribute('size'), 12));
+        const content = columns[0].querySelector('slot[name="content"]').assignedElements()[0];
+        assert.equal(content, pane.querySelector('[slot="content"]'));
+        const actionGroup = columns[1].querySelector('[is="vl-action-group"]');
+        should.exist(actionGroup);
+        const previousAction = actionGroup.querySelector('slot[name="previous-action"]').assignedElements()[0];
+        const nextAction = actionGroup.querySelector('slot[name="next-action"]').assignedElements()[0];
+        assert.equal(previousAction, pane.querySelector('[slot="previous-action"]'));
+        assert.equal(nextAction, pane.querySelector('[slot="next-action"]'));
+      });
+
+      test('de acties elementen krijgen een wizard next/previous attribuut', () => {
+        const pane = fixture('vl-wizard-pane-fixture');
+        const grid = pane.shadowRoot.querySelector('[is="vl-grid"]');
+        const columns = grid.querySelectorAll('[is="vl-column"]');
+        const actionGroup = columns[1].querySelector('[is="vl-action-group"]');
+        const previousAction = actionGroup.querySelector('slot[name="previous-action"]').assignedElements()[0];
+        const nextAction = actionGroup.querySelector('slot[name="next-action"]').assignedElements()[0];
+        assert.isTrue(previousAction.hasAttribute('data-vl-wizard-prev'));
+        assert.isTrue(nextAction.hasAttribute('data-vl-wizard-next'));
+      });
+
+      test('voorziet geen slot elementen indien de acties niet aanwezig zijn', () => {
+        const pane = fixture('vl-wizard-pane-without-actions-fixture');
+        const grid = pane.shadowRoot.querySelector('[is="vl-grid"]');
+        const columns = grid.querySelectorAll('[is="vl-column"]');
+        const actionGroup = columns[1].querySelector('[is="vl-action-group"]');
+        const previousAction = actionGroup.querySelector('slot[name="previous-action"]');
+        const nextAction = actionGroup.querySelector('slot[name="next-action"]');
+        should.not.exist(previousAction);
+        should.not.exist(nextAction);
+      });
+
+      test('de next/previous pane disabled attributen bepalen bij de wizard of er genavigeerd kan worden naar een vorige/volgende pagina', (done) => {
+        // unresolved Promise = next/previous not allowed
+        // resolved Promise = next allowed but previous not allowed
+        // resolved Promise with true boolean = next/previous allowed
```

Comment:
Is het eigenlijk niet handiger om 2 promises te voorzien dan: eentje voor next en eentje voor previous? 
Of eentje, maar dan een object te verwachten als resultaat van de promise ({ next: true, previous: false} om aan te duiden dat next wel mag maar previous niet).

Reply:
Dit is de code van Webuniversum dat we aanroepen, indien gewenst kan je hiervoor een issue maken maar ik vrees dat dergelijke wijzigingen niet zomaar aangenomen gaan worden.

Reply:
Nee, maar we kunnen zelf een andere API exposen en dat vertalen naar een promise voor Webuniversum.

Reply:
Het is niet de bedoeling dat afnemers de callback rechtstreeks gaan zetten, het is puur een API tussen vl-ui-wizard en vl-ui-wizard-pane, maar daarvoor heb ik het wel publiek moeten zetten. De API voor de gebruiker heb ik als boolean voorzien, check enable/disable API op vl-ui-wizard-pane en de data-vl-next-pane-disabled en data-vl-previous-pane-disabled attributen.

Reply:
Ter illustratie, zie de demo implementatie, daar wordt geen callback aangeroepen.

Reply:
Als de callback setter/getter niet private/protected staat (en bijkomend documentatie heeft), kan een afnemer wel het besluit trekken dat dat voor hem bedoeld is. Kunnen we hem dan niet beter private/protected maken?

Reply:
Nee, want dan zou de vl-ui-wizard-pane een protected/private methode van een vl-ui-wizard aanroepen wat not done is. De gebruiker mag gerust de callback gebruiken, al is dat niet nodig. Hopelijk is de demo pagina voldoende informatief, maar wie weet wil de afnemer liever zelf de callback beheren.

Reply:
In Java kan je ook gewoon protected methodes aanroepen vanuit dezelfde package?

Reply:
Offline besloten dat publieke API op zich hier geen probleem is.